src/mantine-hooks/src/use-list-state/use-list-state.ts

```diff
@@ -82,6 +83,13 @@ export function useListState<T>(initialValue: T[] = []): UseListState<T> {
       current.map((item, index) => (condition(item, index) ? fn(item, index) : item))
     );
 
+  const filter = (fn: (item: T) => boolean) => {
+    setState((current) => {
+      const cloned = [...current];
```

Comment:
This operation is redundant, filter method already creates a shallow copy of the array, remove it